创建队列配置guarantee超出集群可用资源导致schedule报错

### Description

提交队列配置guarantee超出集群可用资源导致volcano schedule 失败

### Steps to reproduce the issue

## 1.创建队列 test-1

```yaml
apiVersion: scheduling.volcano.sh/v1beta1
kind: Queue
metadata:
  name: test-1
spec:
  # deserved字段仅用于capacity插件
  deserved:
    cpu: "10"
    memory: 8Gi
  guarantee:
    resource:
      cpu: 1
      memory: 4Gi
  priority: 100
  reclaimable: true
```

## 2.schedule运行正常

## 3.创建队列test-2

```yaml
apiVersion: scheduling.volcano.sh/v1beta1
kind: Queue
metadata:
  name: test-2
  namespace: test
spec:
  # deserved字段仅用于capacity插件
  deserved:
    cpu: "10"
    memory: 8Gi
  guarantee:
    resource:
      cpu: 10000
      memory: 4Gi
  priority: 50
  reclaimable: true
```

## 4.schedule显示Error

```sh
volcano-scheduler-7f666cf77c-gl8mb     0/1     Error     237 (65s ago)   5d6h
```

## 5.报错信息如下

```sh
I0310 09:20:53.573981       1 event_handlers.go:391] Added pod <wx-test/example-df479d598-n7mqj> into cache.
I0310 09:20:53.573991       1 event_handlers.go:391] Added pod <wx-test/example-pod> into cache.
I0310 09:20:53.574003       1 event_handlers.go:391] Added pod <xx-test/example-78764466f5-6c6cp> into cache.
I0310 09:20:53.574012       1 event_handlers.go:391] Added pod <xx-test/example1-78764466f5-h9mxs> into cache.
I0310 09:20:53.574022       1 event_handlers.go:391] Added pod <xx-test/httpbin-868566b7c-krkqq> into cache.
I0310 09:20:53.574036       1 event_handlers.go:391] Added pod <xx-test/nacos-0> into cache.
I0310 09:20:53.574047       1 event_handlers.go:391] Added pod <xx-test/nacos-1> into cache.
I0310 09:20:53.574062       1 event_handlers.go:391] Added pod <xx-test/nacos-2> into cache.
I0310 09:20:53.574071       1 event_handlers.go:391] Added pod <xx-test/nginx-netshoot-1-55cf9b9948-cgxkd> into cache.
I0310 09:20:53.574097       1 event_handlers.go:391] Added pod <xx-test/sae-app-green-example-556d65f9dd-z9ctv> into cache.
I0310 09:20:53.600875       1 cache.go:792] Start metrics collection, metricsConf is map[]
I0310 09:20:53.600892       1 cache.go:797] The interval for querying metrics data is 30s
I0310 09:20:53.600902       1 scheduler.go:90] Scheduler completes Initialization and start to run
W0310 09:20:53.600993       1 node_info.go:231] the argument node is null.
I0310 09:20:53.601051       1 cache.go:1517] The metrics type is not set in the volcano scheduler configmap file. As a result, the CPU and memory load information of the node is not collected.
W0310 09:20:53.601253       1 node_info.go:231] the argument node is null.
W0310 09:20:53.601607       1 node_info.go:231] the argument node is null.
W0310 09:20:53.602044       1 node_info.go:231] the argument node is null.
W0310 09:20:53.602316       1 node_info.go:231] the argument node is null.
W0310 09:20:53.602686       1 node_info.go:231] the argument node is null.
I0310 09:20:53.604091       1 cache.go:1383] There are <1> Jobs, <3> Queues and <6> Nodes in total for scheduling.
I0310 09:20:53.604121       1 session.go:190] Open Session 96d48683-361f-489a-aa1d-0ec61524873d with <1> Job and <3> Queues
E0310 09:20:53.605657       1 runtime.go:77] Observed a panic: resource is not sufficient to do operation: <cpu 164000.00, memory 750343524352.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, nvidia.com/gpu 3000.00, pods 660.00, attachable-volumes-csi-csi-clusterfileplugin 12884901882.00, attachable-volumes-csi-csi.tigera.io 12884901882.00, ephemeral-storage 1648769005709000.00> sub <cpu 10001000.00, memory 8589934592.00>
goroutine 525 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1dfaee0, 0xc00255a450})
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc001c39180?})
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x6b
panic({0x1dfaee0?, 0xc00255a450?})
/usr/local/go/src/runtime/panic.go:770 +0x132
volcano.sh/volcano/pkg/scheduler/util/assert.Assert(0x9a?, {0xc0009af200?, 0xc0005615b8?})
/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:33 +0x165
volcano.sh/volcano/pkg/scheduler/util/assert.Assertf(0x0, {0x230299a?, 0x7f7db6cf7a68?}, {0xc0005615b8?, 0xc0000c4808?, 0xc0025571e0?})
/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:43 +0x4a
volcano.sh/volcano/pkg/scheduler/api.(*Resource).Sub(0xc002558640, 0xc0025584e0)
/go/src/volcano.sh/volcano/pkg/scheduler/api/resource_info.go:260 +0x90
volcano.sh/volcano/pkg/scheduler/plugins/capacity.(*capacityPlugin).OnSessionOpen(0xc002558500, 0xc002192000)
/go/src/volcano.sh/volcano/pkg/scheduler/plugins/capacity/capacity.go:119 +0x8e5
volcano.sh/volcano/pkg/scheduler/framework.OpenSession({0x25e0fb0?, 0xc00020f608?}, {0xc0004b5b90, 0x2, 0x2}, {0x0, 0x0, 0x0})
/go/src/volcano.sh/volcano/pkg/scheduler/framework/framework.go:45 +0x312
volcano.sh/volcano/pkg/scheduler.(*Scheduler).runOnce(0xc000528180)
/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:117 +0x29d
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000ffe6e0, {0x25b4340, 0xc002180000}, 0x1, 0xc00022d920)
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ffe6e0, 0x3b9aca00, 0x0, 0x1, 0xc00022d920)
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.Until(...)
/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
created by volcano.sh/volcano/pkg/scheduler.(*Scheduler).Run in goroutine 139
/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:91 +0x1a7
panic: resource is not sufficient to do operation: <cpu 164000.00, memory 750343524352.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, nvidia.com/gpu 3000.00, pods 660.00, attachable-volumes-csi-csi-clusterfileplugin 12884901882.00, attachable-volumes-csi-csi.tigera.io 12884901882.00, ephemeral-storage 1648769005709000.00> sub <cpu 10001000.00, memory 8589934592.00> [recovered]
panic: resource is not sufficient to do operation: <cpu 164000.00, memory 750343524352.00, hugepages-1Gi 0.00, hugepages-2Mi 0.00, nvidia.com/gpu 3000.00, pods 660.00, attachable-volumes-csi-csi-clusterfileplugin 12884901882.00, attachable-volumes-csi-csi.tigera.io 12884901882.00, ephemeral-storage 1648769005709000.00> sub <cpu 10001000.00, memory 8589934592.00>

goroutine 525 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc001c39180?})
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xcd
panic({0x1dfaee0?, 0xc00255a450?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
volcano.sh/volcano/pkg/scheduler/util/assert.Assert(0x9a?, {0xc0009af200?, 0xc0005615b8?})
	/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:33 +0x165
volcano.sh/volcano/pkg/scheduler/util/assert.Assertf(0x0, {0x230299a?, 0x7f7db6cf7a68?}, {0xc0005615b8?, 0xc0000c4808?, 0xc0025571e0?})
	/go/src/volcano.sh/volcano/pkg/scheduler/util/assert/assert.go:43 +0x4a
volcano.sh/volcano/pkg/scheduler/api.(*Resource).Sub(0xc002558640, 0xc0025584e0)
	/go/src/volcano.sh/volcano/pkg/scheduler/api/resource_info.go:260 +0x90
volcano.sh/volcano/pkg/scheduler/plugins/capacity.(*capacityPlugin).OnSessionOpen(0xc002558500, 0xc002192000)
	/go/src/volcano.sh/volcano/pkg/scheduler/plugins/capacity/capacity.go:119 +0x8e5
volcano.sh/volcano/pkg/scheduler/framework.OpenSession({0x25e0fb0?, 0xc00020f608?}, {0xc0004b5b90, 0x2, 0x2}, {0x0, 0x0, 0x0})
	/go/src/volcano.sh/volcano/pkg/scheduler/framework/framework.go:45 +0x312
volcano.sh/volcano/pkg/scheduler.(*Scheduler).runOnce(0xc000528180)
	/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:117 +0x29d
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:226 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000ffe6e0, {0x25b4340, 0xc002180000}, 0x1, 0xc00022d920)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:227 +0xaf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000ffe6e0, 0x3b9aca00, 0x0, 0x1, 0xc00022d920)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:204 +0x7f
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/src/volcano.sh/volcano/vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:161
created by volcano.sh/volcano/pkg/scheduler.(*Scheduler).Run in goroutine 139
	/go/src/volcano.sh/volcano/pkg/scheduler/scheduler.go:91 +0x1a7
```



### Describe the results you received and expected

null

### What version of Volcano are you using?

master

### Any other relevant information

_No response_

--- a/pkg/scheduler/api/job_info.go
+++ b/pkg/scheduler/api/job_info.go
@@ -574,30 +574,11 @@ func (ji *JobInfo) DeductSchGatedResources(res *Resource) *Resource {
 
 // GetElasticResources returns those partly resources in allocated which are more than its minResource
 func (ji *JobInfo) GetElasticResources() *Resource {
-	elastic := EmptyResource()
 	if ji.Allocated == nil {
-		return elastic
+		return EmptyResource()
 	}
 	minResource := ji.GetMinResources()
-	if ji.Allocated.MilliCPU > minResource.MilliCPU {
-		elastic.MilliCPU = ji.Allocated.MilliCPU - minResource.MilliCPU
-	}
-	if ji.Allocated.Memory > minResource.Memory {
-		elastic.Memory = ji.Allocated.Memory - minResource.Memory
-	}
-
-	for k, v := range ji.Allocated.ScalarResources {
-		if k == v1.ResourcePods{
-			continue
-		}
-		minv := minResource.ScalarResources[k]
-		if v > minv {
-			if elastic.ScalarResources == nil {
-				elastic.ScalarResources = map[v1.ResourceName]float64{}
-			}
-			elastic.ScalarResources[k] = v - minv
-		}
-	}
+	elastic := ExceededPart(ji.Allocated, minResource)
 
 	return elastic
 }

--- a/pkg/scheduler/api/job_info.go
+++ b/pkg/scheduler/api/job_info.go
@@ -572,12 +572,34 @@ func (ji *JobInfo) DeductSchGatedResources(res *Resource) *Resource {
 	return result
 }
 
+// GetElasticResources returns those partly resources in allocated which are more than its minResource
 func (ji *JobInfo) GetElasticResources() *Resource {
-	minResource := ji.GetMinResources()
-	if ji.Allocated.LessEqualPartly(minResource, Zero) {
-		return EmptyResource()
+	elastic := EmptyResource()
+	if ji.Allocated == nil {
+		return elastic
 	}
-	return ji.Allocated.Clone().Sub(minResource)
+	minResource := ji.GetMinResources()
+	if ji.Allocated.MilliCPU > minResource.MilliCPU {
+		elastic.MilliCPU = ji.Allocated.MilliCPU - minResource.MilliCPU
+	}
+	if ji.Allocated.Memory > minResource.Memory {
+		elastic.Memory = ji.Allocated.Memory - minResource.Memory
+	}
+
+	for k, v := range ji.Allocated.ScalarResources {
+		if k == v1.ResourcePods{
+			continue
+		}
+		minv := minResource.ScalarResources[k]
+		if v > minv {
+			if elastic.ScalarResources == nil {
+				elastic.ScalarResources = map[v1.ResourceName]float64{}
+			}
+			elastic.ScalarResources[k] = v - minv
+		}
+	}
+
+	return elastic
 }
 
 func (ji *JobInfo) addTaskIndex(ti *TaskInfo) {

--- a/pkg/scheduler/api/job_info_test.go
+++ b/pkg/scheduler/api/job_info_test.go
@@ -389,6 +389,8 @@ func TestJobInfo(t *testing.T) {
 func TestGetElasticResources(t *testing.T) {
 	resNoGPU := BuildResourceList("1", "1G")
 	resWithGPU := BuildResourceListWithGPU("1", "1G", "1")
+	wantNoGPU := BuildResourceList("1", "1G", []ScalarResource{{Name: "pods", Value: "1"}}...)
+	wantWithGPU := BuildResourceListWithGPU("1", "1G", "1", []ScalarResource{{Name: "pods", Value: "1"}}...)
 	tests := []struct {
 		pods     []*v1.Pod
 		podgroup scheduling.PodGroup
@@ -398,7 +400,7 @@ func TestGetElasticResources(t *testing.T) {
 			pods: []*v1.Pod{
 				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
 			},
-			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
+			podgroup: BuildPodgroup("pg1", "ns1", 1, wantWithGPU),
 			want:     EmptyResource(),
 		},
 		{
@@ -406,24 +408,24 @@ func TestGetElasticResources(t *testing.T) {
 				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
 				buildPod("ns1", "task-2", "node2", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
 			},
-			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
-			want:     NewResource(resNoGPU),
+			podgroup: BuildPodgroup("pg1", "ns1", 1, wantWithGPU),
+			want:     NewResource(wantNoGPU),
 		},
 		{
 			pods: []*v1.Pod{
 				buildPod("ns1", "task-1", "node1", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
 				buildPod("ns1", "task-2", "node2", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
 			},
-			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
-			want:     NewResource(resNoGPU),
+			podgroup: BuildPodgroup("pg1", "ns1", 1, wantWithGPU),
+			want:     NewResource(wantNoGPU),
 		},
 		{
 			pods: []*v1.Pod{
 				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
 				buildPod("ns1", "task-2", "node2", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
 			},
-			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
-			want:     NewResource(resWithGPU),
+			podgroup: BuildPodgroup("pg1", "ns1", 1, wantWithGPU),
+			want:     NewResource(wantWithGPU),
 		},
 	}
 
@@ -433,7 +435,7 @@ func TestGetElasticResources(t *testing.T) {
 			job.AddTaskInfo(NewTaskInfo(pod))
 		}
 		job.SetPodGroup(&PodGroup{PodGroup: test.podgroup})
-		if elastic := job.GetElasticResources(); !reflect.DeepEqual(elastic, test.want) {
+		if elastic := job.GetElasticResources(); !elastic.Equal(test.want, Zero) {
 			t.Fatalf("case %d:expected %+v, got %+v", i, test.want, elastic)
 		}
 	}

--- a/pkg/scheduler/api/job_info_test.go
+++ b/pkg/scheduler/api/job_info_test.go
@@ -385,3 +385,56 @@ func TestJobInfo(t *testing.T) {
 		}
 	}
 }
+
+func TestGetElasticResources(t *testing.T) {
+	resNoGPU := BuildResourceList("1", "1G")
+	resWithGPU := BuildResourceListWithGPU("1", "1G", "1")
+	tests := []struct {
+		pods     []*v1.Pod
+		podgroup scheduling.PodGroup
+		want     *Resource
+	}{
+		{
+			pods: []*v1.Pod{
+				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
+			},
+			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
+			want:     EmptyResource(),
+		},
+		{
+			pods: []*v1.Pod{
+				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
+				buildPod("ns1", "task-2", "node2", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
+			},
+			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
+			want:     NewResource(resNoGPU),
+		},
+		{
+			pods: []*v1.Pod{
+				buildPod("ns1", "task-1", "node1", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
+				buildPod("ns1", "task-2", "node2", v1.PodRunning, resNoGPU, nil, make(map[string]string)),
+			},
+			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
+			want:     NewResource(resNoGPU),
+		},
+		{
+			pods: []*v1.Pod{
+				buildPod("ns1", "task-1", "node1", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
+				buildPod("ns1", "task-2", "node2", v1.PodRunning, resWithGPU, nil, make(map[string]string)),
+			},
+			podgroup: BuildPodgroup("pg1", "ns1", 1, resWithGPU),
+			want:     NewResource(resWithGPU),
+		},
+	}
+
+	for i, test := range tests {
+		job := NewJobInfo("job")
+		for _, pod := range test.pods {
+			job.AddTaskInfo(NewTaskInfo(pod))
+		}
+		job.SetPodGroup(&PodGroup{PodGroup: test.podgroup})
+		if elastic := job.GetElasticResources(); !reflect.DeepEqual(elastic, test.want) {
+			t.Fatalf("case %d:expected %+v, got %+v", i, test.want, elastic)
+		}
+	}
+}

--- a/pkg/scheduler/api/resource_info.go
+++ b/pkg/scheduler/api/resource_info.go
@@ -771,3 +771,17 @@ func (r ResourceNameList) Contains(rr ResourceNameList) bool {
 func IsCountQuota(name v1.ResourceName) bool {
 	return strings.HasPrefix(string(name), "count/")
 }
+
+// ExceededPart returns the partly resource in left which exceed right
+func ExceededPart(left, right *Resource) *Resource {
+	if right == nil {
+		return left
+	}
+
+	if left == nil {
+		return EmptyResource()
+	}
+
+	diff, _ := left.Diff(right, Zero)
+	return diff
+}

--- a/pkg/scheduler/api/test_utils.go
+++ b/pkg/scheduler/api/test_utils.go
@@ -23,6 +23,8 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+
+	"volcano.sh/apis/pkg/apis/scheduling"
 )
 
 func buildNode(name string, alloc v1.ResourceList) *v1.Node {
@@ -90,7 +92,7 @@ type ScalarResource struct {
 	Value string
 }
 
-// BuildResourceList builts resource list object
+// BuildResourceList builds resource list object
 func BuildResourceList(cpu string, memory string, scalarResources ...ScalarResource) v1.ResourceList {
 	resourceList := v1.ResourceList{
 		v1.ResourceCPU:    resource.MustParse(cpu),
@@ -103,7 +105,7 @@ func BuildResourceList(cpu string, memory string, scalarResources ...ScalarResou
 	return resourceList
 }
 
-// BuildResourceListWithGPU builts resource list with GPU
+// BuildResourceListWithGPU builds resource list with GPU
 func BuildResourceListWithGPU(cpu string, memory string, GPU string, scalarResources ...ScalarResource) v1.ResourceList {
 	resourceList := v1.ResourceList{
 		v1.ResourceCPU:    resource.MustParse(cpu),
@@ -115,4 +117,18 @@ func BuildResourceListWithGPU(cpu string, memory string, GPU string, scalarResou
 	}
 
 	return resourceList
+}
+
+// BuildPodgroup builds podgroup
+func BuildPodgroup(name, ns string, minMember int32, minResource v1.ResourceList) scheduling.PodGroup {
+	return scheduling.PodGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+		Spec: scheduling.PodGroupSpec{
+			MinMember:    minMember,
+			MinResources: &minResource,
+		},
+	}
 }

--- a/pkg/scheduler/plugins/capacity/capacity.go
+++ b/pkg/scheduler/plugins/capacity/capacity.go
@@ -116,7 +116,7 @@ func (cp *capacityPlugin) OnSessionOpen(ssn *framework.Session) {
 			if len(queue.Queue.Spec.Guarantee.Resource) != 0 {
 				attr.guarantee = api.NewResource(queue.Queue.Spec.Guarantee.Resource)
 			}
-			realCapability := cp.totalResource.Clone().Sub(cp.totalGuarantee).Add(attr.guarantee)
+			realCapability := api.ExceededPart(cp.totalResource, cp.totalGuarantee).Add(attr.guarantee)
 			if attr.capability == nil {
 				attr.realCapability = realCapability
 			} else {

--- a/pkg/scheduler/plugins/proportion/proportion.go
+++ b/pkg/scheduler/plugins/proportion/proportion.go
@@ -117,7 +117,7 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 			if len(queue.Queue.Spec.Guarantee.Resource) != 0 {
 				attr.guarantee = api.NewResource(queue.Queue.Spec.Guarantee.Resource)
 			}
-			realCapability := pp.totalResource.Clone().Sub(pp.totalGuarantee).Add(attr.guarantee)
+			realCapability := api.ExceededPart(pp.totalResource, pp.totalGuarantee).Add(attr.guarantee)
 			if attr.capability == nil {
 				attr.realCapability = realCapability
 			} else {

